lib/mlir/Dialect/Clift/IR/CliftTypes.cpp

```diff
@@ -179,73 +179,78 @@ using namespace mlir::clift;
 ::mlir::LogicalResult
 EnumAttr::verify(::llvm::function_ref<::mlir::InFlightDiagnostic()> emitError,
                  uint64_t ID,
-                 ::llvm::StringRef name,
-                 mlir::Type element_type,
+                 ::llvm::StringRef,
+                 mlir::Type UnderlyingType,
                  ::llvm::ArrayRef<mlir::clift::EnumFieldAttr> Fields) {
-  if (not element_type.isa<mlir::clift::PrimitiveType>()) {
+  if (not UnderlyingType.isa<mlir::clift::PrimitiveType>())
     return emitError() << "type of enum must be a primitive type";
-  }
-  auto casted = element_type.cast<mlir::clift::PrimitiveType>();
-  if (casted.getKind() != PrimitiveKind::UnsignedKind
-      and casted.getKind() != PrimitiveKind::SignedKind) {
-    return emitError() << "enum underlying type must be a unsigned int";
-  }
 
-  if (casted.getKind() == PrimitiveKind::UnsignedKind) {
-    size_t lastValue = 0;
-    for (auto field : Fields) {
-      auto Max = llvm::APInt::getMaxValue(casted.getSize()).getZExtValue();
-      if (field.getRawValue() > Max) {
-        return emitError() << "enum field " << field.getRawValue()
-                           << "is greater than the max value of the "
-                              "underlying type"
-                           << Max;
-      }
-      if (field.getRawValue() < lastValue) {
-        return emitError() << "enum fields are not in ascending order";
-      }
-      lastValue = field.getRawValue();
-    }
+  const auto PrimitiveType = UnderlyingType.cast<mlir::clift::PrimitiveType>();
```

Comment:
The `isa` is still there, on line 185. I just renamed `element_type` to `UnderlyingType`. We do prefer `PascalCase` for locals, right?

Reply:
Ah sorry, the side-by-side diff confused me because it was much higher in the page.

We do prefer `PascalCase`.
And BTW, thanks for fixing the casing. It was inconsistent.